compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -1367,17 +1367,17 @@ impl<'a, 'll, 'tcx> Builder<'a, 'll, 'tcx> {
     pub fn vector_reduce_fmul(&mut self, acc: &'ll Value, src: &'ll Value) -> &'ll Value {
         unsafe { llvm::LLVMRustBuildVectorReduceFMul(self.llbuilder, acc, src) }
     }
-    pub fn vector_reduce_fadd_algebraic(&mut self, acc: &'ll Value, src: &'ll Value) -> &'ll Value {
+    pub fn vector_reduce_fadd_reassoc(&mut self, acc: &'ll Value, src: &'ll Value) -> &'ll Value {
         unsafe {
             let instr = llvm::LLVMRustBuildVectorReduceFAdd(self.llbuilder, acc, src);
-            llvm::LLVMRustSetAlgebraicMath(instr);
```

Comment:
@saethlin you changed these to "algebraic", but I think they should be just "reassoc" to match what clang does. (And even that is questionable, see https://github.com/rust-lang/stdarch/issues/1533. But that's a separate issue.)

Reply:
Yes, I believe the "algebraic" implementation is technically wrong. But it was an improvement to land because it doesn't explode so dramatically :p